demo/core_13_$.utils.code.js

```diff
@@ -136,3 +136,60 @@ $.utils.code.toSourceSafe = function(value) {
     throw e;
   }
 };
+
+$.utils.code.formatForEval = function(src) {
+  // Eval treats {} as an empty block (return value undefined).
+  // Eval treats {'a': 1} as a syntax error.
+  // Eval treats {a: 1} as block with a labeled statement (return value 1).
+  // Detect these cases and enclose in parenthesis.
+  // But don't mess with: {var x = 1; x + x;}
+  // This is consistent with the console on Chrome and Node.
+  try {
+    var ast = $.utils.acorn.parse(src, { ecmaVersion: 5 });
+  } catch (e) {
+    try {
+      ast = $.utils.acorn.parseExpressionAt(src, 0, { ecmaVersion: 5 });
+      if (src.substring(ast.end).trim() !== '') {
+        throw SyntaxError('Not just an expression');
+      }
+    } catch (e) {
+      // This eval will fail.  Have fun.
+      return src;
+    }
+    // This is an object literal: {'a': 1}
+    return '(' + src + ')';
+  }
+  if (ast.type === 'Program' && ast.body.length === 1 &&
+      ast.body[0].type === 'BlockStatement') {
+    if (ast.body[0].body.length === 0) {
+      // This is an empty object: {}
+      return '(' + src + ')';
+    }
+    if (ast.body[0].body.length === 1 &&
+        ast.body[0].body[0].type === 'LabeledStatement' &&
+        ast.body[0].body[0].body.type === 'ExpressionStatement') {
+      // This is an object literal: {a: 1}
+      return '(' + src + ')';
+    }
+  }
+  return src;
+};
+
+$.utils.code.formatForEval.test = function() {
+  var matrix = {
+    '1 + 2': '1 + 2',
+    '1 + 2;\n3 + 4;': '1 + 2;\n3 + 4;',
+    '{}': '({})',
+    '{};\n3 + 4;': '{};\n3 + 4;',
+    '{"a": 1}': '({"a": 1})',
+    '{"a": 1};\n3 + 4;': '{"a": 1};\n3 + 4;',
+    '{a: 1}': '({a: 1})',
+    '{a: 1};\n3 + 4;': '{a: 1};\n3 + 4;'
+  };
+  for (var key in matrix) {
+    var actual = $.utils.code.formatForEval(key);
+    if (actual !== matrix[key]) {
+      throw Error('Expected: ' + matrix[key] + ' Actual: ' + actual);
```

Comment:
Convention in server tests is always "got:" then "want:".

Reply:
From my experience with DMP, the convention is first/second in Python:
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertEqual

 and actual/expected everywhere else:
https://junit.org/junit4/javadoc/4.12/org/junit/Assert.html
https://www.mathworks.com/help/matlab/ref/matlab.unittest.qualifications.assertable.assertequal.html
https://phpunit.de/manual/6.5/en/appendixes.assertions.html
https://wiki.servoy.com/display/DOCS/JSUnit#JSUnit-assertEquals(expected,actual)

I don't mind what the server does, but teaching the actual/expected terminology to our users sounds like a better plan than inventing new terminology for no reason.

Reply:
My comment was more about order than terminology.

Reply:
JUnit, JSUnit, NUnit, PHPUnit all use expected/actual in that order.  The only package I can find with the reverse order is Matlab.  Sounds like the server's tests ought to be reversed.

Reply:
Server's tests were based on the original golang ones, and the Go style guide specifies the order explicitly.  Given that Go appears to be the odd one out I'm happy to have the in-core tests conform to the usual order and terminology for JS.

I may consider changing the output of the server tests, though I think that the Go stylists actually made the right call here, because it makes it easy to have super-meaningful test output like `Fail: factorial(3) == 5 (want 6)`, which is quite a bit more clear than `Fail: testFactorialThree: expected: 6, actual: 5`.